src/openrct2-ui/windows/Staff.cpp

```diff
@@ -1196,10 +1200,54 @@ void window_staff_overview_tool_down(rct_window* w, rct_widgetindex widgetIndex,
         if (dest_x == LOCATION_NULL)
             return;
 
+        rct_peep* peep = staff_get_peep_from_sprite_id(w->number);
+        if (peep == nullptr)
+            return;
+
         game_do_command(dest_x, 1, dest_y, w->number, GAME_COMMAND_SET_STAFF_PATROL, 0, 0);
+        if (staff_is_patrol_area_set(peep->staff_id, dest_x, dest_y) == true)
+        {
+            s_staff_patrol_area_paint_value = 1;
+        }
+        else
+        {
+            s_staff_patrol_area_paint_value = 0;
+        }
     }
 }
 
+void window_staff_overview_tool_drag(rct_window* w, rct_widgetindex widgetIndex, int32_t x, int32_t y)
+{
+    if (widgetIndex != WIDX_PATROL)
+        return;
+
+    int32_t dest_x, dest_y;
+    footpath_get_coordinates_from_pos(x, y, &dest_x, &dest_y, nullptr, nullptr);
+
+    if (dest_x == LOCATION_NULL)
+        return;
+
+    rct_peep* peep = staff_get_peep_from_sprite_id(w->number);
+    if (peep == nullptr)
+        return;
+
+    bool patrolAreaValue = staff_is_patrol_area_set(peep->staff_id, dest_x, dest_y);
+    if (s_staff_patrol_area_paint_value == 1 && patrolAreaValue == true)
+        return; // Since area is already the value we want, skip...
+    if (s_staff_patrol_area_paint_value == 0 && patrolAreaValue == false)
+        return; // Since area is already the value we want, skip...
```

Comment:
These two statements can be combined by checking if `patrolAreaValue == s_staff_patrol_area_paint_value`. It would make sense to make the static variable a boolean, so you don't have to use magic numbers.